Stopgap fix for #91

CBenoit opened the pull request:
Fix weird context-tag class.

- Remove `Tag::APP_{0-15}`
- Remove `Tag::CTX_{0-15}`
- Remove `Tag::application`
- Remove `Tag::context_specific`
- Rename `Tag::number` into `Tag::inner`
- Rename `ApplicationTag{0-15}` into `ExplicitContextTag{0-15}`
- Rename `ContextTag{0-15}` into `ImplicitContextTag{0-15}`
- Rename `Implicit` into `Optional`
- Add `Tag::is_primitive`
- Add `Tag::application_primitive`
- Add `Tag::application_constructed`
- Add `Tag::context_specific_primitive`
- Add `Tag::context_specific_constructed`
- Add `Tag::number`
- Add `Tag::class`
- Add `Tag::class_and_number`
- Add `Tag::components`
- Add `Tag::is_application`
- Add `Tag::is_context_specific`
- Add `Tag::is_universal`
- Add `Tag::is_private`
- Add `Tag::is_constructed`
- Add `Tag::is_primitive`
- Add `Tag::encoding`
- Add `TagClass`
- Add `Encoding`

cc @Albibek

@RRRadicalEdward this cleans a bit picky internals, but I have one last issue with a test in `picky::x509::wincert`, do you have an idea?
For for details enable `debug_log` feature of `picky-asn1-der` in `picky/Cargo.toml` and run `cargo test`

```
[…]
| debug => | | | | | | | | | | | | | | | | deserialize_newtype_struct: ObjectIdentifierAsn1
| debug => | | | | | | | | | | | | | | | | | deserialize_bytes
| debug => | | | | | | | | | | | | | | | | | | object read: OBJECT IDENTIFIER (len = 9)
| debug => | | | | | | | | | | | | | | | | | | | object buffer: [60, 86, 48, 01, 65, 03, 04, 02, 01]
| debug => | | | | | | | | | | | | | | | | deserialize_unit
| debug => | | | | | | | | | | | | | | | | | object read: NULL (len = 0)
| debug => | | | | | | | | | | | | | | | | | | object buffer: []
| debug => | | | | | | | | | | | | | deserialize_newtype_struct: Optional
| debug => | | | | | | | | | | | | | | deserialize_newtype_struct: ImplicitContextTag0
| debug => | | | | | | | | | | | | | | | CONTEXT_SPECIFIC(00) PRIMITIVE pushed as encapsulator
| debug => | | | | | | | | | | | | | | | deserialize_newtype_struct: Asn1SequenceOf
| debug => | | | | | | | | | | | | | | | | deserialize_seq
| debug => | | | | | | | | | | | | | | | | | tag: SEQUENCE, len: 25
| debug => | | | | | | | | | | | | | | | | | | deserialize_seq
| debug => | | | | | | | | | | | | | | | | | | | tag: OBJECT IDENTIFIER, len: 9
| debug => | | | | | | | | | | | | | | | | | | | | deserialize_seq: INVALID (found OBJECT IDENTIFIER)
| debug => | | | | | | | | | | | | | deserialize_newtype_struct: SignatureAlgorithmIdentifier
| debug => | | | | | | | | | | | | | | deserialize_seq
| debug => | | | | | | | | | | | | | | | tag: UNIVERSAL(0A) CONSTRUCTED, len: 79744507642121
| debug => | | | | | | | | | | | | | | | | deserialize_newtype_struct: ObjectIdentifierAsn1
| debug => | | | | | | | | | | | | | | | | | deserialize_bytes
| debug => | | | | | | | | | | | | | | | | | | object read: BIT STRING (len = 49)
| debug => | | | | | | | | | | | | | | | | | | | object buffer: [0C, 06, 0A, 2B, 06, 01, 04, 01, 82, 37, 02, 01, 04, 30, 7A, 06, 0A, 2B, 06, 01, 04, 01, 82, 37, 02, 01, 0C, 31, 6C, 30, 6A, A0, 46, 80, 44, 00, 64, 00, 65, 00, 63, 00, 6F, 00, 64, 00, 69, 00, 6E]
thread 'x509::wincert::tests::decoding_into_win_certificate' panicked at 'called `Result::unwrap()` on an `Err` value: Message("invalid value: [AlgorithmIdentifier] unsupported algorithm (unknown oid), expected a supported algorithm")', picky/src/x509/wincert.rs:526:105
```

Something is clearly wrong something (notably because of the `tag: UNIVERSAL(0A) CONSTRUCTED, len: 79744507642121`)

RRRadicalEdward commented:
About the falling test. 
It deserializes `SignerInfo::signed_attrs` that is `Optional<ImplicitContextTag0<Attributes>>`. Fails to deserialize an Attribute's oid, stops deserializing `SignerInfo::signed_attrs`, and then tries to deserialize bytes into `SignatureAlgorithmIdentifier`, although the bytes related to the `SignerInfo::signed_attrs`. I continue researching the issue. I will notify you if I find something.
 Also one more test fails - pkcs7::signed_data::tests::decode_with_crl 
![image](https://user-images.githubusercontent.com/55661041/127750859-05dc4463-7bec-4c51-98fd-aacdc62fcab4.png)

RRRadicalEdward commented:
I managed to fix the unit test. I did similar to what you did to SignedData::certificates(encoding constructed as implicit) to  SignerInfo::signed_attrs. Also, I fixed `pkcs7::signed_data::tests::decode_with_crl` unit test. I will push the fixes🙂.

CBenoit commented on picky-asn1-x509/Cargo.toml:
```diff
@@ -17,7 +17,7 @@ repository = "https://github.com/Devolutions/picky-rs"
 [dependencies]
 picky-asn1 = { version = "0.3", path = "../picky-asn1" }
-picky-asn1-der = { version = "0.2", path = "../picky-asn1-der" }
+picky-asn1-der = { version = "0.2", path = "../picky-asn1-der", features = ["debug_log"] }
```
You are right, I forgot to remove it!

CBenoit commented:
@RRRadicalEdward Thank you! :tada: 
I cleaned a bit with your comments and updated CHANGELOG.
If everything is okay with you, you may approve the PR and it’ll automatically get merged :slightly_smiling_face:
